frontend/src/api/index.ts

```diff
@@ -10,13 +11,36 @@ let api: Client | undefined;
 let waiters: Array<Waiter> = [];
 
 export default {
-  init: () => {
+  init: async (actions) => {
     api = new Client(METABACKEND_URL, METABACKEND_NETWORK);
-    api.initialize().then(()=>{
-        // Wake up sleepers waiting for API.
-        for (let i = 0; i < waiters.length; i++) {
-          waiters[i](api);
+    (async () => {
+      try {
+        await api.initialize();
+        actions.afterSuccessfulInitialization();
+      } catch (e) {
```

Comment:
The more I look at this, the less I like my choice of throwing different kinds of exceptions for provider errors. Would another API suit you better?